#3, #4 新規登録及びログイン機能の作成

KS-you opened the pull request:
#3 
#4 

作成したファイル
__init__.py ：機能の設定
models.py：DBの設定
views.py：画面のルーティング、ログイン、新規登録機能の設定
setup.py：アプリの起動設定
requreiment.txt：必要なモジュールの記述

削除ファイル
app.py：setup.pyに変更

パスワードのソルトについてはまだ実装できてません。
確認お願いします。

KS-you committed:
新規登録及びログインに必要なモジュール、DB設計等の設定の記述

mo-ri-regen commented on back/stretch_app/__init__.py:
```diff
@@ -0,0 +1,50 @@
+        'host': 'localhost',
+        'db_name': "Stretch_DB"
+        })
+    # メモリを消費しないためFalseに設定
```
passwordが記載されていますが
envに書かなくてセキュリティ的に問題ない？
それともソルトをあとでつけるから見れてもいいってことですかね？

KS-you committed:
必要なモジュールの追加、変更

mo-ri-regen commented on back/stretch_app/setup.py:
```diff
@@ -0,0 +1,6 @@
+flask_stretch_app = create_app()
+
+if __name__ == '__main__':
+    flask_stretch_app.run(debug=True, host='0.0.0.0', port=5001)
```
flaskがあまりわかっていないんですけど
本番環境にデプロイするときは
ソースを書き換えないといけないのでしょうか？(debugのTrueとFalseを変更する必要あり？)
もしそうなら、書き換える必要がないようにたとえばenvで環境変数に値を持つとかしてほしいです🙏

開発環境でしか通らないのなら対応不要です

mo-ri-regen commented:
@KS-you 
ソースのコメント、ありがとうございます🥰

匙加減の問題ですけど1行ごとにコメント書くの大変だと思うので
パッと見てわかる場合はコメント省略してもらってOKです
(今からわざわざ消す必要はないですが)

KS-you committed:
Merge branch 'develop' into you/#3

KS-you reviewed (commented):
@mo-ri-regen 
コンクリフト解消済みです。

mo-ri-regen commented:
@KS-you 
ありがとうございます
ぼくがコメントしたところ修正されていないようですが
対応中ですかね？

KS-you committed:
Revert "必要なモジュールの追加、変更"

This reverts commit 51d4d09c3f0c512f709fafd29930cecf8bd7991d.

KS-you committed:
__init__.py
・ファイル内のmysqlの設定情報を環境変数に置き換え
・appディレクトリ配置に伴うモジュール名の変更
models.py
・appディレクトリ配置に伴うモジュール名の変更
views.py
・appディレクトリ配置に伴うモジュール名の変更
app.py
・setup.pyから名称の変更
・API起動に伴う設定情報を環境変数に置き換え
html
・APIテスト用のhtmlファイルの追加
css
・APIテスト用のCSSファイルの追加

KS-you commented:
@mo-ri-regen 
確認お願いします。

KS-you committed:
html
・APIテスト用のhtmlファイルの追加

KS-you commented:
@mo-ri-regen 
htmlファイル1つpush忘れです失礼しました。

mo-ri-regen commented on back/app/templates/base.html:
```diff
@@ -9,7 +9,7 @@
             {% endblock title %}
         </title>
-        <link rel="stylesheet" href="{{url_for('static',filename='css/styles.css')}}">
+        <link rel="stylesheet" href="{{url_for('static',filename='css/styels.css')}}">
```
@KS-you 
名前を修正するのではなく誤字を修正したほうがいいと思います
(「styles.css」にしてファイル名を修正する)

KS-you committed:
htmlファイル：css指定ファイル名の修正
cssファイル：styles.cssに変更

KS-you commented:
@mo-ri-regen 
修正しました。

KS-you committed:
Delete back/stretch_app directory

stretch_appは不要なディレクトリなので削除

KS-you commented:
@mo-ri-regen 
不要なディレクトリを削除しました。

mo-ri-regen commented on back/app/models.py:
```diff
@@ -0,0 +1,63 @@
+load_dotenv('/back/.env')
+# .envファイル内のソルト値を変数に代入
+SALT = os.getenv('SALT')
+
```
説明ではペッパー値って言っていたけどソルト？
変数名の問題ならたとえばPEPPER_VALUEにしてもらったほうがいいと思います
SALTが正しいならそのままでOKです(コメントもペッパー値に直してもらったほうがいいですね)

mo-ri-regen commented on back/app/app.py:
```diff
@@ -1,15 +1,13 @@
+load_dotenv()
+DEBUG = os.getenv('DEBUG')
+HOST= os.getenv('HOST')
+PORT = os.getenv('PORT')
```
ほかと合わせて
HOST = os.getenv('HOST')
としてください(スペースいれる)
Black Formatter というのがあるみたいです
→ https://zenn.dev/bellstone/articles/ff5aac74455cbf

KS-you committed:
models.py ペッパー値の環境変数名の変更
app.py HOST環境変数定義の改行の修正

KS-you commented:
@mo-ri-regen 
修正しました。確認お願いします。

mo-ri-regen commented on back/app/models.py:
```diff
@@ -18,7 +18,7 @@
 # .envファイル内のソルト値を変数に代入
-SALT = os.getenv('SALT')
+PEPPER = os.getenv('PEPPER')
 
```
コメントも修正してもらえると...🥺
ソルト値→ペッパー値

KS-you committed:
環境変数のコメントを ソルト値->ペッパー値へ変更
top_pages, youtube_videoes テーブルの定義

KS-you commented:
@mo-ri-regen 
夜分にすいません。
コメントいただいた箇所を修正しました。
合わせて、DBのテーブル定義を修正していますので、確認お願いします。

修正箇所
・環境変数のコメント ソルト値 ->  ペッパー値
・テーブル名の変更 Users -> users
・usersの不要なカラムの削除：
・テーブルの追加：top_pages, youtube_videoes

mo-ri-regen commented on back/app/models.py:
```diff
@@ -29,27 +35,24 @@ def load_user(user_id):
     # カラム定義
-    id = db.Column(db.Integer, primary_key=True, nullable=False)
+    id = db.Column(db.Integer, primary_key=True, nullable=False,)
     # user_nameとmail_addressは検索を早くするためにindex=Trueの追加
```
nullable=False,)
↑カンマがあって動く？
(ほかは末尾カンマがないようなので)

mo-ri-regen commented on back/app/models.py:
```diff
@@ -59,5 +62,35 @@ def validate_password(self, password):
+    id = db.Column(db.Integer, primary_key=True, nullable=False)
+    # user_nameとmail_addressは検索を早くするためにindex=Trueの追加
+    item = db.Column(db.String(1000), nullable=False)
+    url = db.Column(db.String(100), nullable=False)
```
titleカラムがない？

mo-ri-regen commented on back/app/models.py:
```diff
@@ -59,5 +62,35 @@ def validate_password(self, password):
+    my_favorite = db.Column(db.Boolean, default=False, nullable=False)
+    # DateTimeのデフォルト値をユーザー作成時のサーバーの時刻とする
+    created_at = db.Column(db.DateTime, server_default=func.now())
 
```
updated_atがない？

mo-ri-regen commented on back/app/models.py:
```diff
@@ -59,5 +62,35 @@ def validate_password(self, password):
+    created_at = db.Column(db.DateTime, server_default=func.now())
+     # DateTimeのデフォルト値をユーザー作成時のサーバーの時刻とする
+    updated_at = db.Column(db.DateTime, server_default=func.now())
+    videoes = relationship('Top', backref='YouTube')
```
youtube_videoesテーブルはupdated_atがいらないのでは？

mo-ri-regen commented:
@KS-you 
確認しました
仕様と違う部分があるので確認してもらえると🙏

KS-you committed:
models.pyの修正
.gitignore非追跡ディレクトリの追加

KS-you commented:
@mo-ri-regen 
models.py修正しました。
usersテーブルの "," はあっても動作はしましたが削除しました。
修正内容：
・usersテーブルのidカラムの末尾コンマの削除
・youtube_videoesテーブルにtitleカラムの追加
・youtube_videoesテーブルからupdated_atカラムの削除
・top_pagesテーブルにupdated_atカラムの追加

合わせて、データベースのマイグレーションに使用するディレクトリは追跡する必要がないので、
.gitignoreにmigrationsのディレクトリを追加しました確認お願いします。

mo-ri-regen commented:
LGTM
